### PR TITLE
Sparse Voxel Map

### PR DESCRIPTION
**Pull request by cpaxton**
## Motivation and Context

Class from build 3d map example

## Changelog

refactor and clean up 3d mapping code for mobile manipulation

## How Has This Been Tested

real robot hardware tests

## Types of changes

- [X ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist:

- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] I ran `tests/hw_manual_test.py` on hardware and observed that robot behavior looks normal.

### CODE REVIEW

File: src/home_robot/home_robot/mapping/voxel.py [hunk 1]
```diff
@@ -23,3 +26,49 @@ def combine_point_clouds(
...
+        """Add this to our history of observations. Also update the current running map."""
+        assert xyz.shape[-1] == 3
+        assert feats.shape[-1] == self.feature_dim
+        assert xyz.shape[0] == feats.shape[0]
+        self.observations.append((camera_pose, xyz, feats))
+        world_xyz = trimesh.transform_points(xyz, camera_pose)
+
+        if self.xyz is None:
+            self.xyz = world_xyz
+            self.feats = feats
```
Comment (zephirefaith) [inline src/home_robot/home_robot/mapping/voxel.py:51]: there should be an else here right? Voxelization will get rid of duplicate points, but better to be explicit.

Comment (zephirefaith) [reply src/home_robot/home_robot/mapping/voxel.py:51]: Or send in `self.xyz=None, self.feats=None` into `combine_point_clouds` as is and let `combine_point_clouds` handle None checking and skipping unnecessary combination.

File: src/home_robot_hw/home_robot_hw/env/stretch_pick_and_place_env.py [hunk 2]
```diff
@@ -59,9 +65,12 @@ def __init__(
...
 
-        # Create a simple grasp planner object, which will let us pick stuff up.
-        # This takes in a reference to the robot client - will replace "self" with "self.client"
-        self.grasp_planner = GraspPlanner(self, visualize_planner=visualize_planner)
+        if ros_grasping:
+            # Create a simple grasp planner object, which will let us pick stuff up.
+            # This takes in a reference to the robot client - will replace "self" with "self.client"
+            self.grasp_planner = GraspPlanner(self, visualize_planner=visualize_planner)
+        else:
+            self.grasp_planner = None
```
Comment (zephirefaith) [inline src/home_robot_hw/home_robot_hw/env/stretch_pick_and_place_env.py:73]: Can we add a check here which explicitly says that "`visualize_planner` will do nothing as no planner is created" if `ros_grasping==False` but `visualize_planner==True`.

**Review by zephirefaith — CHANGES_REQUESTED** [review]

File: src/home_robot_hw/home_robot_hw/env/stretch_pick_and_place_env.py (hunk 2 above)
Comment (cpaxton) [reply src/home_robot_hw/home_robot_hw/env/stretch_pick_and_place_env.py:73]: done, will throw error

File: src/home_robot/home_robot/mapping/voxel.py (hunk 1 above)
Comment (cpaxton) [reply src/home_robot/home_robot/mapping/voxel.py:51]: yeah this was a bug, just added the else

**Review by zephirefaith — APPROVED** [review]
Thanks for changes! This looks great and super-useful.

**Pull request merged**